draft-nygren-httpbis-httpssvc.md

```diff
@@ -0,0 +1,791 @@
+
+When SvcRecordType is 1, the combination of SvcDomainName and
+SvcFieldValue within each resource record associates an Alternative
+Service Field Value with an origin.
+
+The SvcFieldValue of the HTTPSSVC resource record contains an Alt-Svc
+Field Value, exactly as defined in Section 4 of {{!AltSvc}},
+but with the uri-host moved to the SvcDomainName field.
+
+For example, if the operator of https://www.example.com
+intends to include an HTTP response header like
+
+ Alt-Svc: hq="svc.example.net:8003"; ma=3600, \
+          h2="svc.example.net:8002"; ma=3600
+
+They would also publish an HTTPSSVC DNS RRSet like
+
+ www.example.com. 3600 IN HTTPSSVC 1 svc.example.net. "hq=\\":8003\\" pri=0"
+                          HTTPSSVC 1 svc.example.net. "h2=\\":8002\\" pri=1"
+
+This data type can be represented as an Unknown RR as described in
+{{!RFC3597}}:
+
+ www.example.com. 3600 IN TYPE??? \\# TBD:WRITEME
+
+This construction is intended to be extensible in two ways.  First,
+any extensions that are made to the Alt-Svc format for transmission
+over HTTPS are also applicable here, unless expressly mentioned
+otherwise.
+
+Second, by defining a way to map non-HTTPS schemes and non-default
+ports we provide a way for the HTTPSSVC to be used for them as needed.
+However, by using the origin name for the RRName for scheme https and
+port 443 we allow HTTPSSVC records to be included at the end of CNAME
+chains for existing site implementations without requiring changes in
+the zone containing the origin.
+
+
+# Differences from Alt-Svc as transmitted over HTTP
+
+Publishing an alternative services form HTTPSSVC record in DNS is
+intended to be equivalent to transmitting this field value over HTTPS,
+and receiving an HTTPSSVC record is intended to be equivalent to
+receiving this field value over HTTPS.  However, there are some small
+differences in the intended client and server behavior.
+
+## Omitting Max Age and Persist
+
+When publishing an HTTPSSVC record in DNS, server operators MUST omit the
+"ma" parameter, which encodes the "max age" (i.e. expiration time) of
+an Alt-Svc Field Value.  Instead, server operators SHOULD encode the
+expiration time in the DNS TTL, and MUST NOT set a TTL longer than the
+intended "max age".
+
+When receiving an HTTPSSVC record, clients MAY synthesize a new "ma"
+parameter from the DNS TTL, in order to interoperate with Alt-Svc processing
+subsystems.
+
+When publishing an HTTPSSVC record, server operators MUST omit the
+"persist" parameter, which indicates whether the client should use
+this record on other network paths.  When receiving an HTTPSSVC record,
+clients MUST discard any records that contain a "persist" flag.
+Disabling persistence is important to prevent an adversary on one
+network from implanting a forged DNS record that allows them to
+track users after they leave that network.
+
+## Multiple records and preference ordering
+
+Server operators MAY publish multiple HTTPSSVC records as an RRSET.  When
```

Comment:
Given the requirement to only have one alias (2.4), perhaps it’s worth specifying that this is only applicable to the alt-svc form?

Reply:
Added clarification in #9e5d5a2f60c5410598ca1e9b3c916b75ea452855